include/cuco/hash_functions.cuh

```diff
@@ -16,10 +16,35 @@
 
 #pragma once
 
-#include <cuco/detail/probe_sequence_impl.cuh>
+#include <cuco/detail/hash_functions/murmurhash3.cuh>
+#include <cuco/detail/hash_functions/xxhash.cuh>
+
+#include <cuda/std/type_traits>
 
 namespace cuco {
 
+/**
+ * @brief The 32-bit integer finalizer function of `MurmurHash3` to hash the given argument on host
+ * and device.
+ *
+ * @throw Key type must be 4 bytes in size
+ *
+ * @tparam Key The type of the values to hash
+ */
+template <typename Key>
+using fmix_32 = detail::MurmurHash3_fmix32<Key>;
```

Comment:
Is `fmix_32` the proper short term? I'm inclined to keep its full name for clarity `murmurhash3_fmix32`. Another question: what does `f` stand for here? If it's "finalizer", `fmix` (or finalizer mixer) sounds wordy. Educate me if I'm talking nonsense.

Reply:
It's either "mix function" or "finalizer/mixer". No clue which one is actually correct. That's just how the function was named in the smhasher implementation and everyone I know of are going with this term.

Reply:
There were two[ test failures ](https://gpuci.gpuopenanalytics.com/blue/organizations/jenkins/nvidia%2Fgpuci%2FcuCollections%2Fprb%2FcuCollections-gpu-build/detail/cuCollections-gpu-build/1284/pipeline)with xxhash being the default hasher. Anything we should concern about there?

Reply:
https://github.com/rurban/smhasher/blob/7d6813f00e0a7be3bde9aae4739b1c765d9f5827/MurmurHash3.cpp#L66

Ah `finalization mix`.

I would go with the full term `murmurhash3_fmix32` and `murmurhash3_fmix64` since they are murmur hash3 specific

Reply:
Nope, that was just me not adjusting the unit tests when switching to the new default hasher. I'll rewind that for now and then the tests should pass again (TM)

Reply:
>  would go with the full term murmurhash3_fmix32 and murmurhash3_fmix64 since they are murmur hash3 specific

Maybe `murmurhash3_fmix_32` to follow the same scheme as `xxhash_32` and `murmurhash3_32`? Or is this too snake-case-y?

Reply:
> Maybe murmurhash3_fmix_32 to follow the same scheme as xxhash_32 and murmurhash3_32?

sure, works for me

Reply:
Addressed in [8b1cf87](https://github.com/NVIDIA/cuCollections/pull/310/commits/8b1cf874ca788d121e8a96fb67f3ea4b5e4aad05)